### ISSUE TITLE
[Feature request/Bug fix] InnerBlocks template doesn't update

### ISSUE THREAD

**Issue by gaz-think3** [AdvancedCustomFields/acf#364]
Hey guys,

Loving the new release so far, although I am having an issue with the InnerBlocks template not updating.

Please see example;

Inside my block template

````
<?php 
$columns = get_field('number_of_columns');
$align = get_field('align');
$template = array();

for($i = 0; $columns > $i; $i++):
    array_push($template,  array( 'core/column', array(), array()));
endfor;

?>

<div class="row row__<?=$align?>">
    <h3>Columns Block</h3>
    <InnerBlocks template="<?= esc_attr( wp_json_encode( $template ) ) ?>" templateLock="all" />
</div>
````

I have a range field which lets you set a number between 1 and 12 (the number of columns). I am then using a for loop to build the InnerBlocks template. This does work however, the changes are 'live'. It requires the user to refresh the page. I'm assuming this is something to so do with React not compiling properly.

Is there an easy fix for this or is this not possible? From my side this would be very useful since we are building a custom Columns / Column block for our theme.

As you can see below it does work, although you have to refresh to update the template.

![Screenshot 2020-08-20 at 13 42 25](https://user-images.githubusercontent.com/50641262/90771356-1114f700-e2eb-11ea-8397-eb549263c738.png)

![Screenshot 2020-08-20 at 13 42 02](https://user-images.githubusercontent.com/50641262/90771360-13775100-e2eb-11ea-8e28-dacc51ea982c.png)

**Comment by ather16** [comment]
I've noticed there is no templateLock="false" available ?

**Comment by ather16** [comment]
Can you try and confirm this bug?

**Comment by gaz-think3** [comment]
> Can you try and confirm this bug?

@ather16 It doesn't matter whether it is templateLock="all" or templateLock="false" either way it doesn't work

**Comment by elliotcondon** [comment]
Hi @gaz-think3 

Thanks for the topic and code example. I've set this up locally and can confirm your experience. The InnerBlocks template changes are not reflected in the UI until page reload.

This suggests that the InnerBlocks template property is not something that can be changed after initial setup. I'll add this to our to-do list, and see if I can find a solution in the future.

There are some other limitations with the InnerBlocks component, such as its lack of repeatability within a block, so we may roll our own solution in the near future and solve both issues 👍

**Comment by jessicadesjardins** [comment]
Ran into this problem too... Is there a way to manually trigger a block reload with javascript in order to update its InnerBlocks?

**Comment by elliotcondon** [comment]
Hi @jessicadesjardins 

I'm not sure if this will be possible. Technically, the block is reloaded (JSX is rendered via React) anytime there is a change to the block's data (each time your PHP generated preview HTML is fetched via AJAX).

I believe this may be a current limitation of the `<InnerBlocks />` Component, but am not 100% sure just yet.

**Comment by virgo79** [comment]
Not sure if the problem I'm facing is the same reported here.

I have an `acf-block` that has an `<InnerBlocks />` (Inner block template is: one heading and one  paragraph). 
This block  seems to be frozen when added the first time into the editor.
See the capture screen below:

![acf-block-frozen](https://user-images.githubusercontent.com/6567583/102758808-911f9c00-4373-11eb-89a9-4f4ce42fc87b.gif)

Any changes done into the right panel does not trigger the block refresh.
I've noticed that this happens only if the block is added for first into the document editor.

**Comment by gaz-think3** [comment]
> Not sure if the problem I'm facing is the same reported here.
> 
> I have an `ACF-block` that has an `<InnerBlocks />` (Inner block template is: one heading and one paragraph).
> This block seems to be frozen when added the first time into the editor.
> See the capture screen below:
> 
> ![acf-block-frozen](https://user-images.githubusercontent.com/6567583/102758808-911f9c00-4373-11eb-89a9-4f4ce42fc87b.gif)
> 
> Any changes done into the right panel does not trigger the block refresh.
> I've noticed that this happens only if the block is added for first into the document editor.

@virgo79 Yeah this is the same issue as I was having - as mentioned by @elliotcondon  'There are some...limitations with the InnerBlocks component'

> There are some other limitations with the InnerBlocks component, such as its lack of repeatability within a block, so we may roll our own solution in the near future and solve both issues 👍
